src/test/ui/const-generics/cannot-infer-type-for-const-param.stderr

```diff
@@ -14,7 +14,7 @@ error[E0308]: mismatched types
   --> $DIR/cannot-infer-type-for-const-param.rs:10:22
    |
 LL |     let _ = Foo::<3>([1, 2, 3]);
-   |                      ^^^^^^^^^ expected `Const { ty: usize, val: Unevaluated(DefId(0:18 ~ cannot_infer_type_for_const_param[317d]::main[0]::{{constant}}[0]), []) }`, found `Const { ty: usize, val: Scalar(0x0000000000000003) }`
+   |                      ^^^^^^^^^ expected `3`, found `3usize`
```

Comment:
This... is really weird? I assume the `3` is @oli-obk's printing of same-crate anon consts?

Reply:
I think it's because it can't infer the specific type, rather than an issue with printing. This is a bug, though, which should be fixed by https://github.com/rust-lang/rust/pull/60839. (Which is ready to go as soon as this one is.)

Reply:
I mean that the printing output is from `Unevaluated` for the `3` in `Foo::<3>` and not *the value* `3`.

Reply:
ugh, ok I did not envision user facing effects from that. So.... should I just do `_` or `{{unevaluated}}` or sth?

Reply:
I don't think we need to jump to something else here, but we should consider it carefully.
`_` is clearly the conservative choice, as for "unevaluated" - I don't like that name too much, it just happened to be useful in `ty::Const` (and we could/should pick another name).